packages/trpc/server/routers/viewer/apps.tsx

```diff
@@ -158,18 +160,28 @@ export const appsRouter = router({
             },
           });
 
+          // TODO: This should be done async probably using a queue.
           Promise.all(
             appCredentials.map(async (credential) => {
```

Comment:
Here we could use `for ..of`, right?

Reply:
For of will block the execution. Cal needs something like a queue that handles this operation async and unrelated from the execution of this current function.